dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ProcessRestRepository.java

```diff
@@ -135,6 +150,104 @@ protected void delete(Context context, Integer integer)
         }
     }
 
+    /**
+     * Search method that will take Parameters and return a list of {@link ProcessRest} objects
+     * based on the {@link Process} objects that were in the databank that adhere to these params
+     * @param ePersonUuid   The UUID for the EPerson that started the Process
+     * @param scriptName    The name of the Script for which the Process belongs to
+     * @param processStatusString   The status of the Process
+     * @param pageable      The pageable
+     * @return              A page of {@link ProcessRest} objects adhering to the params
+     * @throws SQLException If something goes wrong
+     */
+    @SearchRestMethod(name = "byProperty")
+    @PreAuthorize("hasAuthority('ADMIN')")
+    public Page<ProcessRest> findProcessesByProperty(@Parameter(value = "userId") UUID ePersonUuid,
+                                                     @Parameter(value = "scriptName") String scriptName,
+                                                     @Parameter(value = "processStatus") String processStatusString,
+                                                     Pageable pageable)
+        throws SQLException {
+        if (StringUtils.isBlank(scriptName) && ePersonUuid == null && StringUtils.isBlank(processStatusString)) {
+            throw new DSpaceBadRequestException("Either a name, user UUID or ProcessStatus should be provided");
+        }
+
+        Context context = obtainContext();
+        EPerson ePerson = epersonService.find(context, ePersonUuid);
+        if (ePersonUuid != null && ePerson == null) {
+            throw new DSpaceBadRequestException("No EPerson with the given UUID is found");
+        }
+
+        ProcessStatus processStatus = StringUtils.isBlank(processStatusString) ? null :
+            ProcessStatus.valueOf(processStatusString);
+        if (processStatus != null && processStatus != ProcessStatus.RUNNING && processStatus != ProcessStatus.COMPLETED
```

Comment:
why do you need to explicitly restrict to these values?
Is this because I mentioned them in the contract, because I simply copied it from https://github.com/DSpace/Rest7Contract/blob/main/processes-endpoint.md#execution-details

Reply:
This explicit check has been removed. Since the actual code still retrieves the status using the [existing enums](https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/content/ProcessStatus.java#L13) this check is being done inherently through that piece of code.
When a status is being searched that does not exist, an IllegalArgumentException is thrown